lib/ipmi_delloem.c

```diff
@@ -338,6 +365,689 @@ usage(void)
+	IPMI_DELL_TAG * tagblock;
+
+		memset (&req,0,sizeof(req));
+
+		req.msg.netfn = IPMI_NETFN_APP;
+		req.msg.lun = 0;
+		req.msg.cmd = IPMI_GET_SYS_INFO;
+		req.msg.data_len = 4;
+		req.msg.data = data;
+		data[0] = 0;		/* get parameter*/
+		data[1] = tag_type;	/* get asset tag*/
+		data[2] = 0;
+		data[3] = 0;
+
+		rsp = intf->sendrecv(intf, &req);
+
+		if (rsp == NULL) {
+			lprintf(LOG_ERR, " Error getting asset tag");
+			return -1;
+		} else if (rsp->ccode > 0) {
+			lprintf(LOG_ERR, " Error getting asset tag: %s",
+				val2str(rsp->ccode, completion_code_vals));
+			return -1;
+		}
+
+		tagblock = (IPMI_DELL_TAG *) (void *) rsp->data;
+		memset(tagstring, 0,tagblock->length);
+		memcpy (tagstring, &tagblock->tag, tagblock->length);
+		tagstring[tagblock->length] = '\0' ;
+
+		return 0;
+
+}
+/*****************************************************************
+* Function Name: 	ipmi_sysinfo_print_board
+*
+* Description:	This function prints FRU Board Area for sysinfo command
+* Input: 		intf 			- pointer to ipmi interface
+*			fru			- pointer to fru area
+*			id			- fru id
+*			offset		- offset to fru area
+*			ipmi_version	- ipmi version
+* Output:
+*
+* Return:
+*
+******************************************************************/
+static void
+ipmi_sysinfo_print_board(struct ipmi_intf * intf, struct fru_info * fru,
+		     uint8_t id, uint32_t offset, uint8_t ipmi_version)
+{
+	char * fru_area;
+	uint8_t * fru_data, *fru_data_tmp;
+	uint8_t language;
+	uint32_t fru_len, area_len, i, j;
+	uint16_t length;
+
+    fru_area=NULL;
+	j = 0;
+
+	i = offset;
+	fru_len = 0;
+
+
+	fru_data_tmp = fru_data = malloc(fru->size + 1);
+	if (fru_data == NULL) {
+		lprintf(LOG_ERR, " Out of memory!");
+		return;
+	}
+	memset(fru_data, 0, fru->size + 1);
+
+	/* read enough to check length field */
+	if (read_fru_area(intf, fru, id, i, 2, fru_data_tmp) == 0)
+//		fru_len = 8 * fru_data[i + 1];
+		fru_len = 8 * fru_data[1];
+	if (fru_len <= 0) {
+		free(fru_data);
+		return;
+	}
+
+	fru_len = 0x200;
+	i = 0;
+	/* read in the full fru */
+	if (read_fru_area(intf, fru, id, i, fru_len, fru_data_tmp) < 0) {
+		free(fru_data);
+		return;
+	}
+	BoardOffsetPtr = (fru_data[BOARD_OFFSET])*OFFSETMULTIPLIER;
+	ProductOffsetPtr = (fru_data[PRODUCT_OFFSET])*OFFSETMULTIPLIER;
+	if(iDRAC_FLAG == IDRAC_14G)
+		language = fru_data[BoardOffsetPtr+BOARDLANGOFFSET];
+	else if (iDRAC_FLAG == IDRAC_13G)
+		language = fru_data[10];
+
+	if (language == 0x00) {
+		printf("Board Language Code : English\n");
+	} else {
+		printf("Board Language Code : Unknown\n");
+	}
+
+	if(iDRAC_FLAG == IDRAC_14G)
+		i=BoardOffsetPtr+BOARDPRODUCTMANFLENOFFSET;
+	else if (iDRAC_FLAG == IDRAC_13G) {
+
+	i = 8;
+
+	i++;	/* skip fru area version */
+	area_len = fru_data[i++] * 8; /* fru area length */
+	i++;	/* skip fru board language */
+	i += 3;	/* skip mfg. date time */
+
+	}
+
+	fru_area = get_fru_area_str(fru_data, &i);
+	if (fru_area != NULL && strlen(fru_area) > 0) {
+		free(fru_area);
```

Comment:
Missing `fru_area = NULL` here.
The same applies to multiple cases below. Pointers MUST be set to NULL immediately after thy are `free()`'d.